[BUG] PGVector does not escape values in an IN (notIn) filter

<!-- Please provide as many details as possible, this will help us to deliver a fix as soon as possible. Thank you! -->

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->
The method `dev.langchain4j.store.embedding.pgvector.PgVectorFilterMapper#formatValuesAsString` does not escape the values of an IN filter and the embedding store search method may fail with an error `java.lang.RuntimeException: org.postgresql.util.PSQLException: Unterminated string literal started at position xxx` if any of the IN values of the metadata filter contains an unescaped '.

This behaviour is unique for the IN and NOT IN filters as the rest of the filter do escape the value with the method `dev.langchain4j.store.embedding.pgvector.PgVectorFilterMapper#formatValue`.

Additionally, insertions do not require to escape the values as they are inserted within Prepared Statements.

**Log and Stack trace**
<!-- Please provide a log and a stack trace (with exception), if applicable. -->

> java.lang.RuntimeException: org.postgresql.util.PSQLException: Unterminated string literal started at position 15,503 in SQL SELECT (2 - (embedding <=> ... ;. Expected  char
	at dev.langchain4j.store.embedding.pgvector.PgVectorEmbeddingStore.search(PgVectorEmbeddingStore.java:340)
	at dev.langchain4j.store.embedding.pgvector.PgVectorEmbeddingStoreIT.test_escape_in(PgVectorEmbeddingStoreIT.java:108)
Caused by: org.postgresql.util.PSQLException: Unterminated string literal started at position 15,503 in SQL SELECT (2 - (embedding <=> ... ;. Expected  char
	at org.postgresql.core.Parser.checkParsePosition(Parser.java:1435)
	at org.postgresql.core.Parser.parseSql(Parser.java:1342)

**To Reproduce**
<!-- Please provide a relevant code snippets to reproduce this bug. -->
```
    @Test
    void test_escape_in() {
        TextSegment[] segments = new TextSegment[] {
                TextSegment.from(
                        "toEscape",
                        Metadata.from(Map.of("text", "This must be escaped '"))),
                TextSegment.from(
                        "notEscape",
                        Metadata.from(Map.of("text", "This does not require to be escaped")))
        };
        List<Embedding> embeddings = new ArrayList<>(segments.length);
        for (TextSegment segment : segments) {
            Embedding embedding = embeddingModel().embed(segment.text()).content();
            embeddings.add(embedding);
        }

        List<String> ids = embeddingStore().addAll(embeddings, Arrays.asList(segments));
        awaitUntilAsserted(() -> assertThat(getAllEmbeddings()).hasSameSizeAs(segments));

        Filter filterEQ = metadataKey("text").isEqualTo("This must be escaped '");

        EmbeddingSearchRequest eqSearchRequest = EmbeddingSearchRequest.builder()
                .maxResults(1)
                .queryEmbedding(embeddings.get(0))
                .filter(filterEQ)
                .build();


        // EqualsTo works as expected
        EmbeddingSearchResult<TextSegment> searchResult = embeddingStore().search(eqSearchRequest);
        EmbeddingMatch<TextSegment> match = searchResult.matches().get(0);
        assertThat(match.score()).isCloseTo(1, withPercentage(1));
        assertThat(match.embeddingId()).isEqualTo(ids.get(0));


        // In filter does not escape values
        Filter filterIN = metadataKey("text").isIn("This must be escaped '");
        EmbeddingSearchRequest inSearchRequest = EmbeddingSearchRequest.builder()
                .maxResults(1)
                .queryEmbedding(embeddings.get(0))
                .filter(filterIN)
                .build();

        searchResult = embeddingStore().search(inSearchRequest);
        match = searchResult.matches().get(0);
        assertThat(match.score()).isCloseTo(1, withPercentage(1));
        assertThat(match.embeddingId()).isEqualTo(ids.get(0));
    }
```

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->

Both searches should work the same way. However, the second one (the `isIn `one) fails with an exception

**Please complete the following information:**
- LangChain4j version: 1.11.0-beta19-SNAPSHOT
- Java version: Java 21

**Additional context**
<!-- Add any other context about the problem here. -->

--- a/langchain4j-pgvector/src/main/java/dev/langchain4j/store/embedding/pgvector/PgVectorFilterMapper.java
+++ b/langchain4j-pgvector/src/main/java/dev/langchain4j/store/embedding/pgvector/PgVectorFilterMapper.java
@@ -144,7 +144,16 @@ abstract class PgVectorFilterMapper {
         }
     }
 
+    String formatCollectionValue(Object value) {
+        if (value instanceof String stringValue) {
+            final String escapedValue = stringValue.replace("'", "''");
+            return '\'' + escapedValue + '\'';
+        } else {
+            return '\'' + value.toString() + '\'';
+        }
+    }
+
     String formatValuesAsString(Collection<?> values) {
-        return "(" + values.stream().map(v -> format("'%s'", v)).collect(Collectors.joining(",")) + ")";
+        return "(" + values.stream().map(this::formatCollectionValue).collect(Collectors.joining(",")) + ")";
     }
 }

--- a/langchain4j-pgvector/src/test/java/dev/langchain4j/store/embedding/pgvector/PgVectorEmbeddingStoreIT.java
+++ b/langchain4j-pgvector/src/test/java/dev/langchain4j/store/embedding/pgvector/PgVectorEmbeddingStoreIT.java
@@ -92,5 +92,18 @@ class PgVectorEmbeddingStoreIT extends EmbeddingStoreWithFilteringIT {
         EmbeddingMatch<TextSegment> match = searchResult.matches().get(0);
         assertThat(match.score()).isCloseTo(1, withPercentage(1));
         assertThat(match.embeddingId()).isEqualTo(ids.get(0));
+
+        // In filter escapes values as well
+        Filter filterNotIN = metadataKey("text").isNotIn("This must be escaped '");
+        EmbeddingSearchRequest notInSearchRequest = EmbeddingSearchRequest.builder()
+                .maxResults(1)
+                .queryEmbedding(embeddings.get(0))
+                .filter(filterNotIN)
+                .build();
+
+        searchResult = embeddingStore().search(notInSearchRequest);
+        match = searchResult.matches().get(0);
+        // It must retrieve the second embedding
+        assertThat(match.embeddingId()).isEqualTo(ids.get(1));
     }
 }

--- a/langchain4j-pgvector/src/test/java/dev/langchain4j/store/embedding/pgvector/PgVectorEmbeddingStoreIT.java
+++ b/langchain4j-pgvector/src/test/java/dev/langchain4j/store/embedding/pgvector/PgVectorEmbeddingStoreIT.java
@@ -1,12 +1,27 @@
 package dev.langchain4j.store.embedding.pgvector;
 
+import static dev.langchain4j.store.embedding.TestUtils.awaitUntilAsserted;
+import static dev.langchain4j.store.embedding.filter.MetadataFilterBuilder.metadataKey;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.data.Percentage.withPercentage;
 import static org.testcontainers.shaded.org.apache.commons.lang3.RandomUtils.nextInt;
 
+import dev.langchain4j.data.document.Metadata;
+import dev.langchain4j.data.embedding.Embedding;
 import dev.langchain4j.data.segment.TextSegment;
 import dev.langchain4j.model.embedding.EmbeddingModel;
 import dev.langchain4j.model.embedding.onnx.allminilml6v2q.AllMiniLmL6V2QuantizedEmbeddingModel;
+import dev.langchain4j.store.embedding.EmbeddingMatch;
+import dev.langchain4j.store.embedding.EmbeddingSearchRequest;
+import dev.langchain4j.store.embedding.EmbeddingSearchResult;
 import dev.langchain4j.store.embedding.EmbeddingStore;
 import dev.langchain4j.store.embedding.EmbeddingStoreWithFilteringIT;
+import dev.langchain4j.store.embedding.filter.Filter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
 import org.testcontainers.containers.PostgreSQLContainer;
 import org.testcontainers.junit.jupiter.Container;
 import org.testcontainers.junit.jupiter.Testcontainers;
@@ -48,5 +63,34 @@ class PgVectorEmbeddingStoreIT extends EmbeddingStoreWithFilteringIT {
     @Override
     protected boolean supportsContains() {
         return true;
+    }
+
+    @Test
+    void test_escape_in() {
+        TextSegment[] segments = new TextSegment[] {
+            TextSegment.from("toEscape", Metadata.from(Map.of("text", "This must be escaped '"))),
+            TextSegment.from("notEscape", Metadata.from(Map.of("text", "This does not require to be escaped")))
+        };
+        List<Embedding> embeddings = new ArrayList<>(segments.length);
+        for (TextSegment segment : segments) {
+            Embedding embedding = embeddingModel().embed(segment.text()).content();
+            embeddings.add(embedding);
+        }
+
+        List<String> ids = embeddingStore().addAll(embeddings, Arrays.asList(segments));
+        awaitUntilAsserted(() -> assertThat(getAllEmbeddings()).hasSameSizeAs(segments));
+
+        // In filter escapes values as well
+        Filter filterIN = metadataKey("text").isIn("This must be escaped '");
+        EmbeddingSearchRequest inSearchRequest = EmbeddingSearchRequest.builder()
+                .maxResults(1)
+                .queryEmbedding(embeddings.get(0))
+                .filter(filterIN)
+                .build();
+
+        EmbeddingSearchResult<TextSegment> searchResult = embeddingStore().search(inSearchRequest);
+        EmbeddingMatch<TextSegment> match = searchResult.matches().get(0);
+        assertThat(match.score()).isCloseTo(1, withPercentage(1));
+        assertThat(match.embeddingId()).isEqualTo(ids.get(0));
     }
 }

